src/35search.js

```diff
@@ -8,1133 +8,1103 @@
 //
 */
 
-function doSearch(databaseid, params, cb) {
-	var res;
-	var stope = {};
-	var fromdata;
-	var selectors = cloneDeep(this.selectors);
-
-	function processSelector(selectors, sidx, value) {
-		//		var val;
-		/*/*		if(sidx == 0) {
-			if(selectors.length > 0 && selectors[0].srchid == 'SHARP') {
-				val = alasql.databases[alasql.useid].objects[selectors[0].args[0]];
-				return processSelector(selectors,sidx+1,val);
-				//selectors.shift();			
-			} else if(selectors.length > 0 && selectors[0].srchid == 'AT') {
-				val = alasql.vars[selectors[0].args[0]];
-				return processSelector(selectors,sidx+1,val);
-				//selectors.shift();
-			} else if(selectors.length > 0 && selectors[0].srchid == 'CLASS') {
-				val = alasql.databases[databaseid].tables[selectors[0].args[0]].data;
-				return processSelector(selectors,sidx+1,val);
-				//selectors.shift();
-				//selectors.unshift({srchid:'CHILD'});
-			} else {
+/**
+	Search class
+	@example
+	SEARCH SUM(/a) FROM ? -- search over parameter object
+*/
 
+yy.Search = class Search {
+	constructor(params) {
+		Object.assign(this, params);
+	}
+
+	toString() {
+		let s = 'SEARCH ';
+		if (this.selectors) s += this.selectors.toString();
+		if (this.from) s += 'FROM ' + this.from.toString();
+		return s;
+	}
+
+	toJS(context) {
+		const s = `this.queriesfn[${this.queriesidx - 1}](this.params,null,${context})`;
+		return s;
+	}
+
+	compile(databaseid) {
+		var dbid = databaseid;
+
+		var statement = (params, cb) => {
+			var res;
+			this.#doSearch(dbid, params, function (data) {
+				res = modify(statement.query, data);
+				if (cb) res = cb(res);
+			});
+			// if(cb) res = cb(res);
+			return res;
+		};
+		statement.query = {};
+		return statement;
+	}
+
+	#doSearch(databaseid, params, cb) {
```

Comment:
What is this `#` doing? Some ES6 tricks I dont know about?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields